app/src/components/main/paginationToolbar/paginationToolbar.jsx

```diff
@@ -8,31 +18,66 @@ import styles from './paginationToolbar.scss';
 
 const cx = classNames.bind(styles);
 
-export const PaginationToolbar = ({
-  activePage,
-  pageCount,
-  pageSize,
-  itemCount,
-  onChangePage,
-  onChangePageSize,
-}) => (
-  <div className={cx('pagination-toolbar')}>
-    {itemCount && <ItemCounter activePage={activePage} pageSize={pageSize} itemCount={itemCount} />}
-    {pageCount > 1 && (
-      <PageButtons activePage={activePage} pageCount={pageCount} onChangePage={onChangePage} />
-    )}
-    {pageSize && <PageSizeControl pageSize={pageSize} onChangePageSize={onChangePageSize} />}
-  </div>
-);
-PaginationToolbar.propTypes = {
-  activePage: PropTypes.number.isRequired,
-  itemCount: PropTypes.number.isRequired,
-  pageCount: PropTypes.number.isRequired,
-  pageSize: PropTypes.number.isRequired,
-  onChangePage: PropTypes.func,
-  onChangePageSize: PropTypes.func,
-};
-PaginationToolbar.defaultProps = {
-  onChangePage: () => {},
-  onChangePageSize: () => {},
+const LAUNCH_PAGE_SIZE_STORAGE_KEY = 'launchPageCount';
+const TEST_PAGE_SIZE_STORAGE_KEY = 'testPageCount';
+const LOG_PAGE_SIZE_STORAGE_KEY = 'logPageCount';
+const OTHER_PAGE_SIZE_STORAGE_KEY = 'otherPageCount';
+
+const PAGE_SIZE_STORAGE_KEYS = {
+  [LEVEL_LAUNCH]: LAUNCH_PAGE_SIZE_STORAGE_KEY,
+  [LEVEL_SUITE]: TEST_PAGE_SIZE_STORAGE_KEY,
+  [LEVEL_TEST]: TEST_PAGE_SIZE_STORAGE_KEY,
+  [LEVEL_STEP]: TEST_PAGE_SIZE_STORAGE_KEY,
+  [LEVEL_LOG]: LOG_PAGE_SIZE_STORAGE_KEY,
+  [LEVEL_NOT_FOUND]: OTHER_PAGE_SIZE_STORAGE_KEY,
 };
+
+export class PaginationToolbar extends Component {
+  static propTypes = {
+    activePage: PropTypes.number.isRequired,
+    itemCount: PropTypes.number.isRequired,
+    pageCount: PropTypes.number.isRequired,
+    pageSize: PropTypes.number.isRequired,
+    onChangePage: PropTypes.func,
+    onChangePageSize: PropTypes.func,
+    userId: PropTypes.string,
+    level: PropTypes.string,
+  };
+
+  static defaultProps = {
+    onChangePage: () => {},
+    onChangePageSize: () => {},
+    userId: '',
+    level: '',
+  };
+
+  state = {
+    pageSize: getPageSize(this.props.userId, PAGE_SIZE_STORAGE_KEYS[this.props.level]),
```

Comment:
Duplicating props in state can lead to unexpected problems.
You can read from local storage in case of no query parameters set (`withPagination.jsx#18-19`)

Reply:
Fixed.